src/sentry/integrations/message_builder.py

```diff
@@ -192,7 +192,10 @@ def build_footer(
         text = rules[0].label if rules[0].label else "Test Alert"
         footer += f" via {url_format.format(text=text, url=rule_url)}"
 
-        if features.has("organizations:slack-block-kit", project.organization):
+        if (
+            features.has("organizations:slack-block-kit", project.organization)
+            and url_format == SLACK_URL_FORMAT
```

Comment:
what is the url format line for?

Reply:
i found out that flipping the feature flag to True causes MsTeams tests to fail because of this line. we're not checking that the message being sent is for Slack

Reply:
oh no, why is this used in msteams? haha

Reply:
this is a `build_footer` function which is used in many places

Reply:
ah a base class and method, got it. If you want to contain the FF stuff, we could just create a Slack specific version, and override the method

Reply:
it's a pure function, so injecting might be harder actually

Reply:
this function is not part of a class though D:

Reply:
> i found out that flipping the feature flag to True causes MsTeams tests to fail because of this line. we're not checking that the message being sent is for Slack

This means it was failing when this was first implemented?

Reply:
it was not failing because the feature flag was not defaulted to True. this means that now, technically, with the feature flag on for everyone, it is failing. so i fixed it

Reply:
Right, but when this FF line was first added, and the flag turned on via the remote location, this was technically breaking, we just didn't realize it :/

Reply:
yep